### ISSUE TITLE
Multiple bots?

### ISSUE THREAD

**Issue by J-o-h-n-M** [nicoschmitt/necrobotvisualizer#10]
First, great project! Really nice to see my bot walking around. I have one issue, I have 6 bots atm, currently it only shows 1 bot. How can i visualize all 6?

**Comment by niicojs** [comment]
That's really not in my plan. But you're welcome to fork and do it.
If you assign correct port in config for each of your bot, you can switch from one to the other changing the config.

**Comment by J-o-h-n-M** [comment]
Ok, i will have a look at it. But a bit stuck as "npm i && npm start" complains that there is no start. I also dont see that defined in package.json. (im pretty new at this)

NVM, update to latest master and it found start.

**Comment by themanticors** [comment]
Just wanted to ask @Muzurio if you had any luck with how to get it to work with multiple bots and if how to do it. I am a noob regarding coding and compiling so if you ELI5 maybe I could get it to work as well. Thanks in advance.

**Comment by J-o-h-n-M** [comment]
@themanticors nope sorry havent gotten around to it to look into it. Im also not that skilled at developing. I am sure someone else will pick this up as so many people are using multiple bots.

**Comment by niicojs** [comment]
First, take some time to explain how you would like it to look like. Then someone else may try to implement it.
Personally, I don't even know how it should work.

**Comment by li81** [comment]
@nicoschmitt U can just let it open the map and then count how many bots are running. Then make a bookmark for each bot like every browser has for a certain website, so we can switch between them and check them out.

**Comment by BinguinBabaja** [comment]
The bookmark thing would be nice!

But maybe a little easier suggestion:
A little dropdown menu to easily switch ports,that you entered in the settings diveded by "," or something

**Comment by degree8th** [comment]
Maybe with using personal unique Websocket port for each tab and bot?

**Comment by theloser** [comment]
I have three bots running, each on a different websocket port. But I can only start one visualizer, so i have to go to settings and change the port everytime i want to check on a different bot. Is there a way to start more than one instance of the visualizer? That would be great.

Edit: nevermind, I somehow can run multiple instances of visualizer, each one connected to a different websocket.

**Comment by li81** [comment]
Where do I change the websocket port?

**Comment by theloser** [comment]
If you are using Necrobot edit config\config.json and change the value of "WebSocketPort". Then start the visualizer, go to settings and edit "Bot Websocket" to your new port (for example: default/first bot = wss://localhost:14251, second bot = wss://localhost:14252, third bot = wss://localhost:14253, ...)

**Comment by li81** [comment]
Didn't work last time, but seems to work now, thx.

**Comment by li81** [comment]
Seems like it's not working for me..

**Comment by li81** [comment]
Is it because I havent logged into the account before?

**Comment by obsess** [comment]
A tab interface would work nicely for this?

**Comment by whiteshark-1975** [comment]
https://gyazo.com/d6895b52bd43f4321906ebfe295ce11b
like this, ope copy 1 set the port press save
open copy 2 press the port and save
in your config of necro set a different port for each bot 14251 14252 14253
like this. problem now saving is general save somewhere globally, if changed in 1 bot its change everywhere, bot will stay on chosen port, but at each clear or restart the same stuff over and over. what we need is to save the port in array() then use a dropdown on the array. or tab the array

**Comment by sinuraya** [comment]
I think the best way to monitor multiple bots is to have a control panel that shows level progress of each bot.  When we click a bot progress,  it will show a bot visualizer.

But when I tried to implement it, I stuck on the message passed by the bot Websocket service. The message is lack of player progress information. It only has username, currency (stardust and poke coin) and other stuff like avatar information. Let's ask necrobot  to provide more information on message they sent through websocket.

**Comment by niicojs** [comment]
I'm actually working on a completly different approach : having a public website you can use where you can pass the bot as parameter. so it can live in the browser and have multiple page open with multiple bot. will be available for testing soon

**Comment by niicojs** [comment]
https://github.com/nicoschmitt/pogo.webui/wiki

**Comment by Kakvyza** [comment]
Can you add command-line support for a port at startup? Example: "NecrobotVisualizer.exe -websocket wss://localhost:14252"
